src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -97,9 +98,24 @@ private void getMealList(HttpServletRequest request, HttpServletResponse respons
         // Every object Meal adds if it contains at least one keyword in at least one field 
         // (type, title, description, ingredients).
         List<Meal> searchedMeal = new ArrayList<>();
+        HashMap<Integer, List<Meal>> sortedMeal = new HashMap<>();
         for (Meal meal : meals) {
-            if (isResultOfSearch(meal, params)) {
-                searchedMeal.add(meal);
+            int frequency = getFrequency(meal, params);
+            if (frequency > 0) {
+                List<Meal> list = new ArrayList<>();
+                if (sortedMeal.get(frequency) != null) {
+                    list = sortedMeal.get(frequency);
+                }
+                list.add(meal);
+                sortedMeal.put(frequency, list);
+            }
+        }
+        for (int i = params.size(); i > 0; i --) {
+            List<Meal> list = sortedMeal.get(i);
+            if (list != null) {
+                for (Meal meal : list) {
+                    searchedMeal.add(meal);
```

Comment:
Use `searchedMeal.addAll(list)` to add all elements of `list` to `searchedMeal` without the `for` loop.